src/main/java/me/itzg/helpers/forge/ForgeInstaller.java

```diff
@@ -163,6 +166,37 @@ private VersionPair extractVersion(Path forgeInstaller) throws IOException {
 
             return new VersionPair(idParts[0], idParts[2]);
         });
+        if (fromVersionJson != null) {
+            return fromVersionJson;
+        }
+
+        return IoStreams.readFileFromZip(forgeInstaller, "install_profile.json", inputStream -> {
+            final ObjectNode parsed = ObjectMappers.defaultMapper()
+                .readValue(inputStream, ObjectNode.class);
+
+            final JsonNode idNode = parsed.path("versionInfo").path("id");
+            if (idNode.isTextual()) {
+                // such as "1.7.10-Forge10.13.4.1614-1.7.10"
+                final String[] idParts = idNode.asText().split("-");
+
+                if (idParts.length == 3) {
```

Comment:
Some of the forge versions only have length 2 here. 1.7.10 sure had an annoying extra bit in the filename.
Here's 1.5.2:
```json
  "versionInfo": {
    "id": "1.5.2-Forge7.8.1.738",
```

Reply:
Great investigation. As I mentioned over in the issue, it would be awesome if you could PR the fix for this. Just reference the same issue as before.

Reply:
```suggestion
                if (idParts.length == 2 || idParts.length == 3) {
```

Reply:
Code suggestions can't be applied after the PR is merged. Can you create a new PR with this?